### PR TITLE
[FEATURE] Store Custom Fields and Remove from PluginStore #169494035

### PR DESCRIPTION
**Pull request by dannyb**
# Overall

Stores the `partreon_id`, `has_nsfw_campaign` in the `user.custom_fields` and removes the data from the `PluginStore` after the user has been created.

## Finishes

* [#169494035](https://www.pivotaltracker.com/story/show/169494035)

### CODE REVIEW

File: plugin.rb
Comment (elskwid) [inline plugin.rb]: This comment seems like it either isn't complete or may be two thoughts. 

> so we can use in during This is called 

just reads weird

File: plugin.rb
Comment (elskwid) [inline plugin.rb]: Is adding a user to a group idempotent?

File: plugin.rb
Comment (elskwid) [inline plugin.rb]: Non-blocking: I could see `Patreon.get("user_info.#{user.email}")` moving to a helper like 

```ruby
def get_user_info_record(email)
    key = "user_info.#{email}"
    Patreon.get(key) || {}
end
```

File: plugin.rb
Comment (elskwid) [inline plugin.rb]: This comment looks incomplete.

**Review by elskwid — CHANGES_REQUESTED** [review]
The comments seem like there might be a little chopped up. Clean those up and this is goog to go.

I like the refactoring.

File: plugin.rb
Comment (dannyb) [reply plugin.rb]: @elskwid yes, `Group.add` and `Group.remove` are idempotent.

https://github.com/discourse/discourse/blob/master/app/models/group.rb#L614
https://github.com/discourse/discourse/blob/master/app/models/group.rb#L641

File: plugin.rb
Comment (dannyb) [reply plugin.rb]: yeah I thought about it, I will do it too

File: plugin.rb
Comment (elskwid) [reply plugin.rb]: Great, thanks.

**Review by elskwid — APPROVED** [review]

**Comment by elskwid** [comment]
@dannyb I've marked this as approved so after the comments are updated you can merge away.

**Pull request merged**
